File: 백준/Silver/2669. 직사각형 네개의 합집합의 면적 구하기/직사각형 네개의 합집합의 면적 구하기.py
import sys
input = sys.stdin.readline

MAX_LEN = 101
numbers = [list(map(int, line.split())) for line in sys.stdin.read().strip().split("\n")]

arr = [[False] * MAX_LEN for _ in range(MAX_LEN)]
cnt = 0

for x1, y1, x2, y2 in numbers:
    for x in range(x1, x2):
        for y in range(y1, y2):
            if arr[x][y]:
                continue
            arr[x][y] = True
            cnt += 1

print(cnt)

File: 백준/Silver/2669. 직사각형 네개의 합집합의 면적 구하기/직사각형 네개의 합집합의 면적 구하기.js
const input = require("fs").readFileSync("/dev/stdin").toString().trim().split("\n")


const numbers = input.slice(0).map((row) => row.split(" ").map(Number))

const answer = new Set()

for (let [x1, y1, x2, y2] of numbers) {
  const graph = Array.from({ length: 101 }, () => Array.from({ length: 101 }, () => false))
  for (let sx = y1; sx <= y2; sx += 1) {
    for (let sy = x1; sy <= x2; sy += 1) {
      graph[sx][sy] = true
      if(graph[sx][sy] && graph[sx][sy-1] && graph[sx-1][sy-1] && graph[sx-1][sy]){
        answer.add(`${sx}${sy}${sx-1}${sy-1}`)
      }

    }
  }
}

console.log(answer.size)